File: BankKata/BankCore/Models/Account.cs
using BankCore.Repositories;
using System;

namespace BankCore.Models
{
    public class Account
    {
        private TransactionRepository transactionRepository;

        public Account(TransactionRepository transactionRepository)
        {
            this.transactionRepository = transactionRepository;
        }

        public void Deposit(int amount)
        {
            transactionRepository.AddDeposit(amount);
        }

        public void Withdraw(int amount)
        {
            
        }

        public void PrintStatement()
        {
            
        }
    }
}

File: BankKata/BankCoreVBNetEdition/Account.vb
Public Class Account
    ReadOnly _transactionRepository As TransactionRepository
    ReadOnly _statementPrinter As StatementPrinter

    Sub New(transactionRepository As TransactionRepository, statementPrinter As StatementPrinter)
        _transactionRepository = transactionRepository
        _statementPrinter = statementPrinter
    End Sub

    Sub Deposit(amount As Integer)
        _transactionRepository.AddDeposit(amount)
    End Sub

    Sub Withdraw(amount As Integer)
        _transactionRepository.AddWithdrawal(amount)
    End Sub

    Sub PrintStatement()
        _statementPrinter.Print(_transactionRepository.AllTransactions())
    End Sub
End Class
